homeassistant/components/media_player/ps4.py

```diff
@@ -0,0 +1,341 @@
+                save_json(file, cred)
+                hass.components.configurator.request_done(
+                    _CONFIGURING.pop(host))
+            except OSError:
+                _LOGGER.error("Could not save credentials file")
+            _CONFIGURING[host] = configurator.request_config(
+                NOTIFICATION_TITLE_PAIR, connect_callback,
+                description="On PS4 go to settings > Mobile App Settings > Add Device\
+                and enter PIN that is displayed",
+                submit_caption='Enter',
+                fields=[{
+                    'id': 'pin',
+                    'name': 'PIN',
+                    'type': 'string'}]
+            )
+
+    if host not in _CONFIGURING:
+        credentials = _creds
+
+    _CONFIGURING[host] = configurator.request_config(
+        NOTIFICATION_TITLE, configuration_callback,
+        description="Press start to begin config. Then\
+        in PS4 Second Screen App, refresh devices and select 'PS4-Waker'",
+        submit_caption='Start'
+    )
+
+
+def get_credentials(hass, config):
+    """Get credentials if any."""
+    host = config.get(CONF_HOST)
+    file = hass.config.path(CONFIG_FILE)
+    try:
+        creds = load_json(file)
+        credentials = creds[host]
+    except KeyError:
+        credentials = None
+        _LOGGER.debug("No credentials found for %s", (host))
+    return credentials
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the PS4 platform."""
+    import ps4python3 as pyps4
+    host = config.get(CONF_HOST)
+    creds = None
+    creds = get_credentials(hass, config)
+    if creds is not None:
+        if host in _CONFIGURING:
+            hass.components.configurator.request_done(_CONFIGURING.pop(host))
+        name = config.get(CONF_NAME)
+        ps4 = pyps4.Ps4(host, creds)
+        games_file = hass.config.path(GAMES_FILE)
+        add_devices([PS4Device(name, host, ps4, games_file)], True)
+    elif creds is None:
+        _creds = pyps4.Credentials()
+        request_configuration(hass, config, add_devices, _creds)
+        return True
+
+
+class PS4Device(MediaPlayerDevice):
+    """Representation of a PS4."""
+
+    def __init__(self, name, host, ps4, games_file):
+        """Initialize the ps4 device."""
+        self._ps4 = ps4
+        self._host = host
+        self._name = name
+        self._state = STATE_UNKNOWN
+        self._games_filename = games_file
+        self._media_content_id = None
+        self._media_title = None
+        self._media_image = None
+        self._source = None
+        self._source_selected = None
+        self._games = self.load_games()
+        self._source_list = list(sorted(self._games.values()))
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Retrieve the latest data."""
+        try:
+            status = self._ps4.get_status()
+            state = status.get('status')
```

Comment:
There is no reson to store this in a local variable that I can see.

Reply:
> status = self._ps4.get_status()
This is an api call which communicates with a PS4 over LAN and returns a dict. Multiple states for the component are retrieved from that dict. My preference is to limit the # of api calls.

>  state = status.get('status')
Perhaps. If you would post an alternative, Ill look into it.